core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -82,24 +81,39 @@ public Joiner(JoinerConfig conf) {
   public void configurePipeline(MultiInputPipelineConfigurer pipelineConfigurer) {
     MultiInputStageConfigurer stageConfigurer = pipelineConfigurer.getMultiInputStageConfigurer();
     Map<String, Schema> inputSchemas = stageConfigurer.getInputSchemas();
-    FailureCollector collector = init(inputSchemas,
-                                      pipelineConfigurer.getMultiInputStageConfigurer().getFailureCollector());
+    FailureCollector collector = pipelineConfigurer.getMultiInputStageConfigurer().getFailureCollector();
+    init(inputSchemas, collector);
     collector.getOrThrowException();
-    //validate the input schema and get the output schema for it
-    stageConfigurer.setOutputSchema(getOutputSchema(inputSchemas, collector));
+    if (!conf.inputSchemasAvailable(inputSchemas) && !conf.containsMacro(conf.OUTPUT_SCHEMA) &&
```

Comment:
nit: `JoinerConfig.OUTPUT_SCHEMA`

Reply:
Can we move this check to init as well?

Reply:
Fixed `conf.OUTPUT_SCHEMA` -> `JoinerConfig.OUTPUT_SCHEMA`.
The check doesn't belong in init because init is called from:
1. here, i.e. configurePipeline
2. prepareRun
3. initialize

Making this check at configure time is sufficient; IMO we don't need to check this again in other places.